File: 2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.cpp
// Last updated: 3/13/2026, 8:16:31 PM
class Solution {
public:
    vector<int> buildArray(vector<int>& nums) {
        vector<int> result;
        for(int index = 0 ; index < nums.size() ; index++){
            result.push_back(nums[nums[index]]);
        }
        return result;
    }
};

File: 2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.java
// Last updated: 3/13/2026, 8:16:27 PM
class Solution {
    public int[] buildArray(int[] nums) {
        int[] arr = new int[nums.length];
        for(int i = 0 ; i < nums.length ; i++){
            arr[i] = nums[nums[i]];
        }
        return arr;
    }
}